internal/discord/commands.go

```diff
@@ -112,7 +112,153 @@ func (d *Discord) Purge(s *discordgo.Session, i *discordgo.InteractionCreate) {
 //	duration:	string
 //	reason:		string
 func (d *Discord) Exile(s *discordgo.Session, i *discordgo.InteractionCreate) {
-	return
+	isFirst := true
+	optionMap := mapOptions(i)
+	exileRole := d.Roles[i.GuildID]["Exiled"]
+	verifiedRole := d.Roles[i.GuildID]["Verified"]
+	logMsg, err := d.LogCommand(i.Interaction)
+	if err != nil {
+		fmt.Printf("Failed to log: %v\n", err)
+	}
+
+	// Calculate duration of exile
+	startTime := time.Now()
+	duration, err := parseDuration(optionMap["duration"].StringValue())
+	if err != nil {
+		fmt.Printf("Failed to parse duration: %v\n", err)
+		return
+	}
+
+	userToExile := optionMap["user"].UserValue(nil)
+
+	// Check if user exists in guild
+	memberToExile, err := d.CheckUserInGuild(i.GuildID, userToExile.ID)
+	if err != nil {
+		tempstr := fmt.Sprintf("Could not find user <@%v> in guild", userToExile.ID)
+		fmt.Printf("%v: %v\n", tempstr, err)
+		RespondToInteraction(s, i.Interaction, tempstr, &isFirst)
+		return
+	}
+
+	// Check if user already has exile role
+	isExiled := false
+	for _, role := range memberToExile.Roles {
+		if role == exileRole.ID {
+			isExiled = true
+		}
+	}
+	if isExiled {
+		tempstr := fmt.Sprintf("User <@%v> is already exiled", userToExile.ID)
+		RespondToInteraction(s, i.Interaction, tempstr, &isFirst)
+		logMsg.Embeds[0].Description += fmt.Sprintf("\n%v", tempstr)
+		_, err = d.Session.ChannelMessageEditEmbed(d.ModLoggingChannelID, logMsg.ID, logMsg.Embeds[0])
+		if err != nil {
+			fmt.Printf("Unable to edit log message: %v\n", err)
+		}
+		return
+	}
+
+	// Exile the user
+	// Remove Verified role
+	err = s.GuildMemberRoleRemove(i.GuildID, userToExile.ID, verifiedRole.ID)
+	if err != nil {
+		tempstr := fmt.Sprintf("Could not remove role %v from user <@%v>", verifiedRole.Name, userToExile.ID)
+		fmt.Printf("%v: %v\n", tempstr, err)
+		RespondToInteraction(s, i.Interaction, tempstr, &isFirst)
+
+		logMsg.Embeds[0].Description += fmt.Sprintf("\n%v", tempstr)
+		_, err = d.Session.ChannelMessageEditEmbed(d.ModLoggingChannelID, logMsg.ID, logMsg.Embeds[0])
+		if err != nil {
+			fmt.Printf("Unable to edit log message: %v\n", err)
+		}
+		return
+	}
+	// Add exiled role
+	err = s.GuildMemberRoleAdd(i.GuildID, userToExile.ID, exileRole.ID)
+	if err != nil {
+		tempstr := fmt.Sprintf("Could not give user <@%v> role %v", userToExile.ID, exileRole.Name)
+		fmt.Printf("%v: %v\n", tempstr, err)
+		RespondToInteraction(s, i.Interaction, tempstr, &isFirst)
+
+		logMsg.Embeds[0].Description += fmt.Sprintf("\n%v", tempstr)
+		_, err = d.Session.ChannelMessageEditEmbed(d.ModLoggingChannelID, logMsg.ID, logMsg.Embeds[0])
+		if err != nil {
+			fmt.Printf("Unable to edit log message: %v\n", err)
+		}
+		return
+	}
```

Comment:
since we're going to use this logic twice (once here once in `Unexile`, let's make this a helper function for both where we remove one role, and add the other. params could be `roleToRemove` and `roleToAdd` and then add/remove in a general sense